internalshared/configutil/userlockout.go

```diff
@@ -0,0 +1,171 @@
+package configutil
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/go-secure-stdlib/parseutil"
+	"github.com/hashicorp/hcl"
+	"github.com/hashicorp/hcl/hcl/ast"
+)
+
+const (
+	UserLockoutThresholdDefault    = 5
+	UserLockoutDurationDefault     = 15 * time.Minute
+	UserLockoutCounterResetDefault = 15 * time.Minute
+	DisableUserLockoutDefault      = false
+)
+
+type UserLockout struct {
+	Type                   string
+	LockoutThreshold       int64         `hcl:"-"`
+	LockoutThresholdRaw    interface{}   `hcl:"lockout_threshold"`
+	LockoutDuration        time.Duration `hcl:"-"`
+	LockoutDurationRaw     interface{}   `hcl:"lockout_duration"`
+	LockoutCounterReset    time.Duration `hcl:"-"`
+	LockoutCounterResetRaw interface{}   `hcl:"lockout_counter_reset"`
+	DisableLockout         bool          `hcl:"-"`
+	DisableLockoutRaw      interface{}   `hcl:"disable_lockout"`
+}
+
+func ParseUserLockouts(result *SharedConfig, list *ast.ObjectList) error {
+	var err error
+	result.UserLockouts = make([]*UserLockout, 0, len(list.Items))
+	userLockoutsMap := make(map[string]*UserLockout)
+	for i, item := range list.Items {
+		var userLockoutConfig UserLockout
+		if err := hcl.DecodeObject(&userLockoutConfig, item.Val); err != nil {
+			return multierror.Prefix(err, fmt.Sprintf("userLockouts.%d:", i))
```

Comment:
It doesn't look like we're using `multierror.Append` in this function. Would it make more sense to return using `fmt.Errorf`?

Reply:
I followed ParseListeners function and others defined in internalshared/configutil, they follow this without multierror.Append . This will  just prefix userlockouts(config name - all/userpass/ldap/approle) with error.

Reply:
Got it, thanks. This was mostly a nit so if this is a pattern we use frequently elsewhere then that makes sense.